src/components/Transcripts.tsx

```diff
@@ -85,19 +106,128 @@ function LoadedTranscripts({ transcripts: unsorted }: {
             </Select>
           }
         </Flex>
+        <Button onClick={handleNameMapModal} variant='brand' >用户匹配</Button>
         <Spacer />
         <Button variant="ghost" rightIcon={<ChevronRightIcon />}
           isDisabled={transcriptIndex == 0}
           onClick={() => replaceUrlParam(router, "transcriptId", sorted[transcriptIndex - 1].transcriptId)}        
         >后一次</Button>
-      </Flex>
+      </Flex> 
       {!summary ? <Loader /> :
         <Card variant="outline" backgroundColor="backgroundLight">
           <CardBody>
             <ReactMarkdown>{summary.summary}</ReactMarkdown>
           </CardBody>
         </Card>
       }
+      {nameMapModal &&
+        <NameMapModal
+          nameMap={nameMap}
+          groupUsers={groupUsers}
+          onClose={() => {
+            handleNameMapModal();
+            refetchSummary();
+            refetchNameMap();
+          }} />}
     </Flex>
   );
 }
+
+function NameMapModal({ nameMap, groupUsers, onClose }: {
```

Comment:
Move this function into a separate file under the components folder. Also break this function down to smaller functions. It's too long as it stands right now.